### PR TITLE
Add serviceport var to CRD

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
This patch allows users to modify the nfd-master Service port via CRD
with the variable Operand.ServicePort , will deafult to 12000 when not
defined

Signed-off-by: Carlos Eduardo Arango Gutierrez <carangog@redhat.com>

### CODE REVIEW

File: controllers/nodefeaturediscovery_controls.go [hunk 1]
```diff
@@ -35,6 +37,8 @@ type ResourceStatus int
 const (
 	Ready ResourceStatus = iota
 	NotReady
+
+	servicePort int = 12000
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:41]: Suggest slight renaming:

```suggestion
	defaultServicePort int = 12000
```

File: config/samples/nfd.kubernetes.io_v1_nodefeaturediscovery.yaml [hunk 2]
```diff
@@ -8,6 +8,7 @@ spec:
     namespace: node-feature-discovery-operator
     image: gcr.io/k8s-staging-nfd/node-feature-discovery:master
     imagePullPolicy: Always
+    servicePort: 24000
```
[the remark this answers is not in the document]
Comment (ArangoGutierrez) [reply config/samples/nfd.kubernetes.io_v1_nodefeaturediscovery.yaml:11]: Yup I was going for a different value to show how it works, doing something diff from default, but same as default works for me

File: controllers/nodefeaturediscovery_controls.go [hunk 3]
```diff
@@ -305,6 +309,15 @@ func DaemonSet(n NFD) (ResourceStatus, error) {
...
 	}
 
+	// update nfd-master service port
+	if n.ins.Spec.Operand.ServicePort != 0 && obj.ObjectMeta.Name == "nfd-master" {
+		portFlag := fmt.Sprintf("--port=%d", n.ins.Spec.Operand.ServicePort)
+		obj.Spec.Template.Spec.Containers[0].Args = []string{portFlag}
+	} else if obj.ObjectMeta.Name == "nfd-master" {
+		portFlag := fmt.Sprintf("--port=%d", defaultServicePort)
+		obj.Spec.Template.Spec.Containers[0].Args = []string{portFlag}
+	}
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:319]: Hmm, my comment regarding this was lost somewhere yesterday 🤔  Anyway, I somehow find this more difficult to read than it should. What do you think about something like this (could probably be made polished even more but the main point is checking for "nfd-master" only once)

```suggestion
        if obj.ObjectMeta.Name == "nfd-master" {
                port := defaultServicePort
                if n.ins.Spec.Operand.ServicePort != 0 {
                        port = n.ins.Spec.Operand.ServicePort
                }
                portFlag := fmt.Sprintf("--port=%d", port)
                obj.Spec.Template.Spec.Containers[0].Args = []string{portFlag}
        }

```

**Review by marquiz — APPROVED** [review]
Nice improvement @ArangoGutierrez!

/label tide/merge-method-squash
/lgtm

**Pull request merged**